### PR TITLE
Initial support for multiple MediaStreams in a single WebRtcConnection

### PR DESCRIPTION
**Pull request by jcague**
**Description**

This PR adds initial support for handling multiple streams in a single WebRtcConnection inside Erizo.
[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

Not needed.

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: erizo/src/erizo/WebRtcConnection.cpp [hunk 1]
```diff
@@ -385,7 +398,18 @@ void WebRtcConnection::onTransportData(std::shared_ptr<DataPacket> packet, Trans
   if (getCurrentState() != CONN_READY) {
     return;
   }
-  media_stream_->onTransportData(packet, transport);
+  char* buf = packet->data;
+  RtpHeader *head = reinterpret_cast<RtpHeader*> (buf);
+  RtcpHeader *chead = reinterpret_cast<RtcpHeader*> (buf);
+  uint32_t ssrc = head->getSSRC();
+  if (chead->isRtcp()) {
+    ssrc = chead->getSourceSSRC();
```
Comment (lodoyun) [inline erizo/src/erizo/WebRtcConnection.cpp]: I think this won't work for sender reports. In that case, we need `chead->getSSRC()`

File: erizo/src/erizo/WebRtcConnection.cpp [hunk 2]
```diff
@@ -385,7 +398,18 @@ void WebRtcConnection::onTransportData(std::shared_ptr<DataPacket> packet, Trans
...
+  }
+  forEachMediaStream([packet, transport, ssrc] (const std::shared_ptr<MediaStream> &media_stream) {
+    if (media_stream->isSourceSSRC(ssrc) || media_stream->isSinkSSRC(ssrc)) {
```
Comment (lodoyun) [inline erizo/src/erizo/WebRtcConnection.cpp]: This is ok. `isSourceSSRC` would make sure packets from a publisher are properly routed while `isSink` would work for feedback packets from a subscriber.

File: erizo/src/erizo/WebRtcConnection.cpp [hunk 3]
```diff
@@ -201,8 +211,10 @@ bool WebRtcConnection::processRemoteSdp() {
   extension_processor_.setSdpInfo(local_sdp_);
   local_sdp_->updateSupportedExtensionMap(extension_processor_.getSupportedExtensionMap());
 
-  local_sdp_->video_ssrc_list.push_back(media_stream_->getVideoSinkSSRC());
-  local_sdp_->audio_ssrc = media_stream_->getAudioSinkSSRC();
+  forEachMediaStream([this] (const std::shared_ptr<MediaStream> &media_stream) {
+    local_sdp_->video_ssrc_list.push_back(media_stream->getVideoSinkSSRC());
+    local_sdp_->audio_ssrc = media_stream->getAudioSinkSSRC();
```
Comment (lodoyun) [inline erizo/src/erizo/WebRtcConnection.cpp]: This is not part of this PR but, do you have a plan here? The obvious solution is to have another list for audio ssrcs.

Comment (jcague) [reply erizo/src/erizo/WebRtcConnection.cpp]: I'd rather move the ssrc tracking to a new class called stream_sdp or similar.
